### PR TITLE
Wikitext docs overhaul first step

### PR DESCRIPTION
**Pull request by pmario**
- The WikiText tiddler overview tiddler shows a logical sort order stored in the field `user-story`
   - The list is ordered by complexity -- simple -> complex
   - The list is 4 columns and is responsive with 2 columns for small screens
- "Formatting in WikiText" uses a table now and has more info
- The PR moves the multi-colums CSS definition used in "Formatting List Results as Tables with CSS - Specified Columns Methods" to the $:/_tw5.com-styles tiddler, so it can be reused 
- Markdown is a new definition tiddler. I don't want to have external links, when we explain our TW syntax.
- "Widgets in Wikitext" ... links to Widgets now, which shows the list of all widgets. There is no need to have that list while explaining the wikitext syntax.

There may be more steps, but this could be a start. This [discussion at Talk](https://talk.tiddlywiki.org/t/a-better-documentation-for-wikitext/4066) triggered the PR.

### CODE REVIEW

**Comment by pmario** [comment]
ZIP file to play with: [wikitext-overhaul-3.zip](https://github.com/Jermolene/TiddlyWiki5/files/9213559/wikitext-overhaul-3.zip)

**Comment by pmario** [comment]
bump

File: editions/tw5.com/tiddlers/concepts/WikiText.tid [hunk 1]
```diff
@@ -1,15 +1,18 @@
 created: 20131205155227468
-modified: 20140919191220377
+modified: 20220728185721771
 tags: Concepts Reference
 title: WikiText
 type: text/vnd.tiddlywiki
+user-story: [[Headings in WikiText]] [[Paragraphs in WikiText]] [[Hard Linebreaks in WikiText]] [[Formatting in WikiText]] [[Dashes in WikiText]] [[Block Quotes in WikiText]] [[Lists in WikiText]] [[Definitions in WikiText]] [[Horizontal Rules in WikiText]] [[Tables in WikiText]] [[Code Blocks in WikiText]] [[Typed Blocks in WikiText]] [[Images in WikiText]] [[Linking in WikiText]] [[Variables in WikiText]] [[Macro Definitions in WikiText]] [[Macro Calls in WikiText]] [[Macros in WikiText]] [[Transclusion in WikiText]] [[Transclusion and Substitution]] [[Styles and Classes in WikiText]] [[HTML in WikiText]] [[Widgets in WikiText]] [[WikiText Parser Modes]]
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/concepts/WikiText.tid:6]: It looks like the demo works by modifying the docs tiddler itself; that's bad because it triggers the dirty state (we had the same problem recently with the GettingStarted tiddler). It would be better to use a temporary tiddler, just using this field to copy from if the field is missing or blank.

Comment (Jermolene) [reply editions/tw5.com/tiddlers/concepts/WikiText.tid:6]: Alternatively, perhaps it is not necessary to make the ordering editable? In any event, I don't think that ordering the docs is sufficient; we also need to introduce group headings or something to show the gross structure.

**Review by Jermolene — COMMENTED** [review]
Thanks @pmario a couple of comments

**Comment by pmario** [comment]
>Alternatively, perhaps it is not necessary to make the ordering editable? 

I'll remove the draggable macro. 

>In any event, I don't think that ordering the docs is sufficient; we also need to introduce group headings or something to show the gross structure.

It would be possible to create something that is similar to https://tiddlywiki.com/plugins/tiddlywiki/tw2parser/#Markup 

But have a look at the code of that tiddler. If any users opens that tiddler the only feedback we get is: "WTF .. That's way to complicated." ... and off they go. That code is a monster. 

It has to be significantly simpler. ... **I'll think about it.**

**Comment by pmario** [comment]
@Jermolene ... 

- There is a new Cheet-Sheet now, that is listed at the end of  the WikiText tiddler. 
- The "Wikitext Cheet-Sheet" is "self contained" according to the macros it uses to build the table
  - There is an "Export Cheet-Sheet" button, which export the tiddler as JSON + all it's dependencies
  - The dependencies are tagged: Wikitext Cheet-Sheet
- If the JSON file is imported into an Empty HTML all the links in the table turn into external-links to tiddlywiki.com
- If the cheet-sheet is part of tiddlywiki.com all the links are internal

This it the exported file to play with: [Wikitext Cheet-Sheet.zip](https://github.com/Jermolene/TiddlyWiki5/files/9396470/Wikitext.Cheet-Sheet.zip)

The wikitext used in the file is intentionally "old style" ... I want to work with older wikis too. 

The cool thing is ... once users get hold of the mechanism, it's easy to extend the info for them self. The table has some extra whitespace for readability with a monospace font. It looks OKish in the TW default editor. 

What do you think?

PS: I did also add an HTML Entity tiddler to the PR, which is _not_ part of the JSON file

**Comment by pmario** [comment]
@Jermolene .. This should be ready to merge and I think it may be of value for new users. Also the discussion at Talk: https://talk.tiddlywiki.org/t/elephant-in-the-room/4782/11

File: editions/tw5.com/tiddlers/concepts/WikiText.tid [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 
 ~WikiText can also be inserted to the text field using the [[Editor toolbar]].
 
-See [[Formatting text in TiddlyWiki]] for an introduction to WikiText.
+@@.multi-columns
+<<list-links filter:"[tag[WikiText]]">>
+@@
 
-The following elements of WikiText syntax are built into the core:
+!! Cheet Sheet
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/concepts/WikiText.tid:17]: The correct spelling is "Cheat Sheet", I'm afraid!

File: editions/tw5.com/tiddlers/system/tw5.com-styles.tid [hunk 3]
```diff
@@ -197,3 +198,24 @@ div.content {
...
+/* WikiText rules */
+
+.multi-columns,
+.fourcolumns {
+  display: block;
+  column-count: 4;
+  column-gap: 1em;
+}
+
+/* make `code` elements less distracting */
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/system/tw5.com-styles.tid:211]: This sounds like a somewhat opinionated visual change. In such situations it is much easier to review the PR if you can include before and after screenshots.

Also please note that our coding style is to use sentence case comments (ie the first letter should be capitalised).

File: editions/tw5.com/tiddlers/wikitext/HTML Entities.tid [hunk 4]
```diff
@@ -0,0 +1,60 @@
+created: 20130204165019000
+modified: 20220822133445149
+tags: WikiText
+title: HTML Entities
+type: text/vnd.tiddlywiki
+
+!! Summary
+
+Use HTML entities to enter characters that can't easily be typed on an ordinary keyboard. They take the form of an ampersand (`&`), an identifying string, and a terminating semi-colon (`;`), e.g. `&amp;`.
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/HTML Entities.tid:9]: It is clear not to use contractions like "can't", but instead to spell them out in full eg "cannot". (I've made a note to clarify the docs on this https://tiddlywiki.com/#Spelling)

**Review by Jermolene — COMMENTED** [review]
Thanks @pmario – I can't try this out until later today, but a couple of comments in the meantime.

Just to emphasise that adding cheatsheets is a very welcome improvemen!

**Comment by pmario** [comment]
@Jermolene .. I did update the PR according to your comments

**Comment by pmario** [comment]
@Jermolene ... please consider this PR for a merge. ... I'd like to link the Cheat-Sheat to this article once it's published. https://gnulinux.ch/serie-strukturierte-notizen-tiddlywiki

File: editions/tw5.com/tiddlers/concepts/WikiText.tid [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
-See [[Formatting text in TiddlyWiki]] for an introduction to WikiText.
+@@.multi-columns
+<<list-links filter:"[tag[WikiText]]">>
+@@
 
-The following elements of WikiText syntax are built into the core:
+!! Cheat Sheet
 
-<<list-links "[tag[WikiText]]">>
+* The [[Wikitext Cheat-Sheet]] will give you a overview about all the ~TiddlyWiki markup.
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/concepts/WikiText.tid:19]: I do feel bad @pmario – I'm afraid that "Cheat-Sheet" is never hyphenated in English.

Both "Cheatsheet" and "Cheat sheet" are equally valid English. I think we should settle on "Cheat sheet" because it is clearer for non-English speakers.

We should capitalise  "Sheet" in headings (as you have done above).

Another issue is that all tiddler titles that include the word "Wikitext" capitalise the "T" so it is "WikiText". It's ugly, and seems less appropriate now that we would like de-emphasise camel case, but it is more important to be consistent than anything else.

File: editions/tw5.com/tiddlers/howtos/Formatting List Results as Tables with CSS - Specified Columns Methods.tid [hunk 6]
```diff
@@ -17,13 +17,9 @@ For other table-making techniques see also:
...
 /* FOUR COLUMN MODE */
 .fourcolumns {
-   display:block;
-   column-count:4;
-   column-gap:1em;
-   -moz-column-count:4;
-   -moz-column-gap:1em;
-   -webkit-column-count: 4;
-   -webkit-column-gap:1em;
+   display: block;
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/howtos/Formatting List Results as Tables with CSS - Specified Columns Methods.tid:20]: It's not clear why these changes belong in this PR.

File: editions/tw5.com/tiddlers/system/tw5.com-styles.tid [hunk 7]
```diff
@@ -197,3 +198,24 @@ div.content {
...
+
+.multi-columns,
+.fourcolumns {
+	display: block;
+	column-count: 4;
+	column-gap: 1em;
+}
+
+/* Tone down `code` elements */
+.no-code-border code {
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/system/tw5.com-styles.tid:212]: Rather than toning them down, I'd much rather that we lost the borders entirely, and just styled the table cells to look like code elements. These nested borders look awful:

File: editions/tw5.com/tiddlers/system/tw5.com-styles.tid [hunk 8]
```diff
@@ -197,3 +198,24 @@ div.content {
 .tc-btn-download:active {
 	background: #1475ff;
 }
+
+/* WikiText rules */
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/system/tw5.com-styles.tid:202]: One big problem is that the formatting breaks down entirely when viewed at mobile sizes. I really think that it is essential that the cheat sheet is usable on a small screen.

File: editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+The following table contains the basic ~TiddlyWiki wikitext features. The "Cheat-Sheet" and all dependencies can be exported using the button below.
+
+<<exportMeButton "Export Cheat-Sheet">>
+
+!! Cheat Sheet
+
+| Example                        | Example Markup                 |h
+| how the example markup renders | what to enter as tiddler text  |
+|>| ! <<.xL "Basic Text Formatting" "Formatting in WikiText">>    |
+|''bold''                        |`''bold''` (tow single hyphens) |
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid:54]: "tow" is a typo.

I think the entire comment `(two single hyphens)` doesn't belong here. Either we add another column for "Notes" or we should drop it.

File: editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid [hunk 10]
```diff
@@ -1,19 +1,38 @@
...
-* `''bold''` for ''bold text'' (<<.icon $:/core/images/bold>>)<br><br>
-* `//italic//` for //italic text// (<<.icon $:/core/images/italic>>)<br><br>
-* `__underscore__` for __underscored text__ (<<.icon $:/core/images/underline>>)<br><br>
-* `^^superscript^^` for ^^superscripted^^ text (<<.icon $:/core/images/superscript>>)<br><br>
-* `,,subscript,,` for ,,subscripted,, text (<<.icon $:/core/images/subscript>>)<br><br>
-* `~~strikethrough~~` for ~~strikethrough~~ text (<<.icon $:/core/images/strikethrough>>)
+<div class="no-code-border">
+
+|Wikitext | Button |Win Shortcut | Mac |Rendered Output |h
+|2 single hyphens are used for `''bold text''`| <kbd><<.icon $:/core/images/bold>></kbd> | <kbd>ctrl-B</kbd> |<kbd>win-B</kbd> |2 single hyphens are used for ''bold text''|
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid:13]: It seems confusing to include editor shortcuts here. Maybe it would make sense in the cheat sheet itself, but here it just seems out of place (we don't do anything similar in any of the other wikitext docs).

File: editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid [hunk 11]
```diff
@@ -1,19 +1,38 @@
...
+|2 commas are used for `,,subscripted,,` text | <kbd><<.icon $:/core/images/subscript>></kbd> | <kbd>ctrl-shift-B</kbd> |<|2 commas are used for ,,subscripted,, text |
+|2 tilde signs are used for `~~strikethrough~~` text | <kbd><<.icon $:/core/images/strikethrough>></kbd> | <kbd>ctrl-T</kbd> |<|2 tilde signs are used for ~~strikethrough~~ text |
+|1 backtick is used for ```code` `` | <kbd><<.icon $:/core/images/mono-line>></kbd> | <kbd>ctrl-M</kbd> |<|1 backtick is used for `code` |
+
+''Special case to show embedded backticks''
+
+|Wikitext |Rendered Output|h
+|<code>&#96;&#96;double backticks allows &#96;embedded&#96; backticks&#96;&#96;</code>|``double backticks allows `embedded` backticks``|
+
+''For those who are interested''
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid:26]: We don't phrase things like that. I've mentioned this before: these are reference documents. They are not supposed to be idiomatic and colloquial, they are supposed to formal and crisp.

Better to say "Further Information" or "Advanced Topics".

File: editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid [hunk 12]
```diff
@@ -1,19 +1,38 @@
...
+|2 slashes are used for `//italic text//`| <kbd><<.icon $:/core/images/italic>></kbd> | <kbd>ctrl-I</kbd> |<kbd>win-I</kbd> |2 slashes are used for //italic text//|
+|2 underscores are used for `__underscored text__`| <kbd><<.icon $:/core/images/underline>></kbd> | <kbd>ctrl-U</kbd> |<kbd>win-U</kbd> |2 underscores are used for __underscored text__ |
+|2 circumflex accents are used for `^^superscripted^^` text | <kbd><<.icon $:/core/images/superscript>></kbd> | <kbd>ctrl-shift-P</kbd> |<|2 circumflex accents are used for ^^superscripted^^ text |
+|2 commas are used for `,,subscripted,,` text | <kbd><<.icon $:/core/images/subscript>></kbd> | <kbd>ctrl-shift-B</kbd> |<|2 commas are used for ,,subscripted,, text |
+|2 tilde signs are used for `~~strikethrough~~` text | <kbd><<.icon $:/core/images/strikethrough>></kbd> | <kbd>ctrl-T</kbd> |<|2 tilde signs are used for ~~strikethrough~~ text |
+|1 backtick is used for ```code` `` | <kbd><<.icon $:/core/images/mono-line>></kbd> | <kbd>ctrl-M</kbd> |<|1 backtick is used for `code` |
+
+''Special case to show embedded backticks''
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid:21]: Shouldn't these be a heading for semantic reasons?

File: editions/tw5.com/tiddlers/wikitext/HTML Entities.tid [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+| &amp;para; | &para; | paragraph sign |~| &amp;raquo; | &raquo; | right angle quote |
+| &amp;sect; | &sect; | section sign |~| &amp;times; | &times; | multiplication symbol |
+| &amp;uarr; | &uarr; | up arrow |~| &amp;darr; | &darr; | down arrow |
+| &amp;larr; | &larr; | left arrow |~| &amp;rarr; | &rarr; | right arrow |
+| &amp;lArr; | &lArr; | double left arrow |~| &amp;rArr; | &rArr; | double right arrow |
+| &amp;harr; | &harr; | left right arrow |~| &amp;hArr; | &hArr; | double left right arrow |
+
+!! Accented Characters
+
+The table below shows how accented characters can be built up by subsituting the underscore (_) into the corresponding character:
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/HTML Entities.tid:50]: Sadly I don't think that this table is at all clear:

image

I think you're trying to say that the underscore should be replaced by the letter that you want to add to the accent, but we're not showing enough information to make that clear.

File: editions/tw5.com/tiddlers/wikitext/Widgets in WikiText.tid [hunk 14]
```diff
@@ -1,11 +1,11 @@
 caption: Widgets
 created: 20131205160840915
-modified: 20220513113347606
+modified: 20220728190445097
 tags: WikiText
 title: Widgets in WikiText
 type: text/vnd.tiddlywiki
 
-Widgets provide rich functionality within WikiText. They have the same syntax as [[HTML elements|HTML in WikiText]], but the tag name always starts with `$`. For example:
+Widgets provide rich functionality within WikiText. They have the same syntax as HTML elements, but the tag name always starts with `$`. For example:
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Widgets in WikiText.tid:8]: Why did you remove the link to "HTML in WikiText"? All the information in that tiddler is relevant for widgets.

File: editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid [hunk 15]
```diff
@@ -0,0 +1,143 @@
+created: 20220821184022101
+modified: 20221010130927409
+tags: WikiText
+title: Wikitext Cheat-Sheet
+type: text/vnd.tiddlywiki
+
+\define extLink(text tid) [ext[$text$|https://tiddlywiki.com/#$tid$]]
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid:7]: Why is the cheat sheet so special that it needs it's own export button? We really need to strive for consistency.

Surely it would make more sense to just arrange a tag pill that can be dragged to export the cheat sheet to another wiki?

File: editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+
+<<exportMeButton "Export Cheat-Sheet">>
+
+!! Cheat Sheet
+
+| Example                        | Example Markup                 |h
+| how the example markup renders | what to enter as tiddler text  |
+|>| ! <<.xL "Basic Text Formatting" "Formatting in WikiText">>    |
+|''bold''                        |`''bold''` (tow single hyphens) |
+|//italics//                     |`//italics//`          |
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid:55]: Why is some of the formatting pulled in via transclusion, and some of it typed directly?

File: editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid [hunk 17]
```diff
@@ -0,0 +1,143 @@
...
+[[$(me)$]] [tag[$(me)$]]
+\end
+
+\define exportMeButton(btnText:"Export Me & Tagged")
+\whitespace trim
+<$set name="me" value=<<currentTiddler>> >
+  <$tiddler tiddler="$:/core/templates/exporters/JsonFile">
+    <$button class="tc-btn-download">
+      <$action-sendmessage $message="tm-download-file" $param=<<currentTiddler>> exportFilter=<<exportFilter>> filename={{{ [<me>addsuffix{!!extension}] }}}/>
+      <$text text="""$btnText$"""/>
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid:20]: I do not think we should add any new macros that use textual substitution. Here's there's clearly an alternative.

File: editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid [hunk 18]
```diff
@@ -0,0 +1,143 @@
...
+\whitespace trim
+<$set name="link" value={{{ [[$tid$]is[tiddler]then<link """$text$""" """$tid$""">else<extLink """$text$""" """$tid$""">] }}}>
+<<link>>
+</$set>
+\end
+
+\define .t(tid) <$transclude tiddler="""$tid$""" mode=block/>
+\define .cb(title) <$codeblock code={{$title$}}/>
+
+<!-- Helper macros. The "old" wikitext style is intentional. The code should be compatible with TW v5.1.15 -->
```
Comment (Jermolene) [inline editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid:36]: I don't think that backwards compatibility should be a goal, not if it constrains the design to the extent that it has. Let's do it properly using modern features.

**Review by Jermolene — COMMENTED** [review]
Thanks @pmario I pulled down the PR to look at it properly.

I am afraid there's quite a long way to go before this can be merged. Sorry not to have been able to give all the feedback in one go.

File: editions/tw5.com/tiddlers/system/tw5.com-styles.tid (hunk 8 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/system/tw5.com-styles.tid:202]: **I'll create a completely new PR for this issue.** 

That's a general problem with TW tables. This problem has always been there, the PR only makes it visible. 

There are 2 main reasons.

 1. For small screens we need:  `word-wrap` to be set to `word-wrap: anywhere` within tables. 
 2. default margins for `blockquotes, OL-LI and UL-li` need to be smaller in tables too. Especially if they are nested as shown in the examples here. They create too much white space.

File: editions/tw5.com/tiddlers/system/tw5.com-styles.tid (hunk 7 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/system/tw5.com-styles.tid:212]: >These nested borders look awful:

That's very subjective. On small screens there is a large gap between the top and the bottom of the table. It's only on a PC where it looks strange. 

Hmmm, We may need a different setting for big and small screens :/  Some more testing will be needed

File: editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid (hunk 10 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid:13]: >It seems confusing to include editor shortcuts here. 

I think, it's a big improvement. All the changes made for this PR have their origin in posts from Talk TiddlyWiki. There are a lot of complains about missing wikitext info for starters. 

>Maybe it would make sense in the cheat sheet itself, but here it just seems out of place 

There isn't enough space on mobile screens. In this tiddler it works, except keyboard shortcuts don't work on mobile. So they are read only there. 

>(we don't do anything similar in any of the other wikitext docs).

That's right. I think, we should add more info about keyboard shortcuts, where it belongs to. But having 10 tiddlers more in this PR I would probably need a year to get it merged. ... 

IMO it's the -docs branch. We should iterate fast here. If something doesn't work, removing it is simple. Adding and formatting costs a lot of time.

But we didn't even test it with real end-users. We didn't promote the new docs, so we didn't get any new feedback.  -- I personally don't need a single line of all this docs added here.

File: editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid (hunk 12 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/wikitext/Formatting in WikiText.tid:21]: Yes -- but am I allowed to use H3s ... without any H2s prior in the text.

File: editions/tw5.com/tiddlers/wikitext/HTML Entities.tid (hunk 13 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/wikitext/HTML Entities.tid:50]: I'll add some more examples and rephrase the text

File: editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid (hunk 15 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid:7]: >Why is the cheat sheet so special that it needs it's own export button?

It should be easy for users to export the basic stuff and **extend the info with their own docs**. ... It's an attempt to get more feedback and it may even be possible that someone sends us an improved JSON file that we could include into the docs, without the need, that we need to do the work.

It would be possible to make the cheat-sheet a plugin. -- but users won't improve our plugins, they just complain that info is missing.

File: editions/tw5.com/tiddlers/howtos/Formatting List Results as Tables with CSS - Specified Columns Methods.tid (hunk 6 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/howtos/Formatting List Results as Tables with CSS - Specified Columns Methods.tid:20]: The style is hardcoded with a `<style>` attribute in this tiddler and it's not globally available. 

I do need the `.fourcolumns` style definition in a different tiddler whis is part of this PR. That's why this definition was moved to the `editions/tw5.com/tiddlers/system/tw5.com-styles.tid` stylesheet also part of this PR, to make it globally available.

File: editions/tw5.com/tiddlers/system/tw5.com-styles.tid (hunk 7 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/system/tw5.com-styles.tid:212]: >These nested borders look awful:

They would look much better, if https://github.com/Jermolene/TiddlyWiki5/pull/6912 would have been merged, which fixes those problems.

File: editions/tw5.com/tiddlers/system/tw5.com-styles.tid (hunk 8 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/system/tw5.com-styles.tid:202]: see:  **fix table overflow for small screens** #7010

File: editions/tw5.com/tiddlers/system/tw5.com-styles.tid (hunk 7 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/system/tw5.com-styles.tid]: did remove it from this PR and create a new issue, so the info isn't forgotten. The new issue is more generic. https://github.com/Jermolene/TiddlyWiki5/issues/7011

File: editions/tw5.com/tiddlers/wikitext/HTML Entities.tid (hunk 13 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/wikitext/HTML Entities.tid:50]: Now looks like this: 

image

File: editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid (hunk 16 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/wikitext/Wikitext Cheat-Sheet.tid:55]: There are 2 goals

- Wherever possible I use plain text
- If not possible want to show the users, how formatting in tables or lists can be achieved

With some of the first iterations I did try to use custom styles to make the content work. But it got so complicated, that I wasn't able to handle it, without extra stylesheet definitions.

File: editions/tw5.com/tiddlers/concepts/WikiText.tid (hunk 5 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/concepts/WikiText.tid:19]: >Both "Cheatsheet" and "Cheat sheet" are equally valid English. I think we should settle on "Cheat sheet" because it is clearer for non-English speakers.

I'm a non English speaker and I don't like `Cheat sheet` for the following 2 reasons. 

At the moment I use a tag **Wikitext Cheat-Sheet**, which would be changed to `Wikitext Cheat sheet` if I apply your suggestion. -- I don't like that. As a tag I would prefer `Wikitext Cheatsheet` which imo would be consistent with eg `$:/tags/Stylesheet` 

At the moment the "details" tiddlers are named eg: `Cheat-Sheet Transclusion` which would be `Cheat sheet Transclusion` which imo doesn't look good as a tiddler name. I'd prefer `Cheatsheet Transclusion`

@Jermolene ... Would this be OK for you too?

Comment (pmario) [reply editions/tw5.com/tiddlers/concepts/WikiText.tid:19]: Since "Cheetsheet" is valid, I'll go with it. Reasoning see my last post.

File: editions/tw5.com/tiddlers/howtos/Formatting List Results as Tables with CSS - Specified Columns Methods.tid (hunk 6 above)
Comment (pmario) [reply editions/tw5.com/tiddlers/howtos/Formatting List Results as Tables with CSS - Specified Columns Methods.tid:20]: - Neither https://caniuse.com/?search=column-gap
- nor https://caniuse.com/?search=column-count  

need a prefix anymore. So I think it can be safely removed

**Comment by Jermolene** [comment]
> It would be possible to make the cheat-sheet a plugin. -- but users won't improve our plugins, they just complain that info is missing.

Good suggestion! On reflection I think it is essential that the cheatsheet be packaged as a plugin, perhaps using integration hooks to plug itself into the editor to make it easily accessible during the writing process.

**Comment by pmario** [comment]
>Good suggestion! On reflection I think it is essential that the cheatsheet be packaged as a plugin, perhaps using integration hooks to plug itself into the editor to make it easily accessible during the writing process.

OK. Should I make it a core-plugin?  

With a more docs it would be possible to get rid of the big-blue Save-Button by linking to the plugin info. 
It should be possible to tell the users how to add their own content and save it as a bundle if they need to.

**Comment by pmario** [comment]
>>OK. Should I make it a core-plugin?

>Yes, but I'd be cautious about investing too much work without discussion.

What do you mean with "without discussion"? 

I would want to make it a Cheasheet edition, where the GettingStarted points to a GettingStarted in the plugin. ... or .. where the $:/DefaultTiddlers opens the "Wikitext Cheatsheet" tiddler ... 

IMO it should also be straight forward to create a "static version" of the Cheatsheet for easy linking with a fast page load. .. But that would be a completely new PR

>Which big blue save button do you want to get rid of?

image

>> It should be possible to tell the users how to add their own content and save it as a bundle if they need to.

>Remember that "bundle" is not a term that we use in the core except in a descriptive sense.

OK. How should we call the whole process of saving a JSON file from a predefined filter from the Advanced search tiddler as shown in the image below. .. That's basically how bundles are created, if they contain more than 1 tiddler.

image
